Add policy for target platforms reaching EOL before a ROS distro (#6231)
ros2/ros2_documentation: source/The-ROS2-Project/Platform-EOL-Policy.rst

.. _PlatformEOLPolicy:

Platform EOL Policy
===================

.. contents:: Table of Contents
   :depth: 1
   :local:

> Every :doc:`ROS distribution <Releases>` has **target platforms**, such as Windows 11 or Ubuntu 24.04.

Reviewer: ```suggestion
Every :doc:`ROS distribution <Releases>` has **target platforms**, also known as operating systems, such as Windows 11 or Ubuntu 24.04.
```

> Every :doc:`ROS distribution <Releases>` has **target platforms**, also known as operating systems, such as Windows 11 or Ubuntu 24.04.

Reviewer 2: ```suggestion
Every :doc:`ROS distribution <Releases>` has **target platforms**, also known as operating systems, such as Windows 11 or Ubuntu 24.04.
Our policy is to not support platforms beyond their official end of life.
```

Author: I simplified the starting section in [ab5fa7e](https://github.com/ros2/ros2_documentation/pull/6231/commits/ab5fa7e138bb410b5d9ddff63d49f3501fb344d3) to make it both include the key point, and call out the intended audience

This page explains what happens when a target platform reaches its own end-of-life (EOL) before the ROS distribution reaches EOL.

What to expect
--------------

When a target platform reaches EOL:

> * All ROS distributions immediately drop support for that platform.

Reviewer: I'm struggling for a software way to say this a bit softer but I'm coming up short. Maybe a brief explanation would help?

Author: I tried shortening it to "stop supporting" in dd1b3059. How does that sound?

> * Existing ROS packages will remain functional, but they will not be updated.

Reviewer: ```suggestion
* Existing ROS packages will remain available and functional, but they will no longer be updated.
```

When a target platform reaches EOL, it stops receiving critical bug and security fixes.

> ROS distributions drop support immediately so that the ROS buildfarm may remove those runners to protect itself from potential unpatched security vulnerabilities.

Reviewer: ```suggestion
To protect ourselves from potentially unpatched security vulnerabilities, we proactively drop all EOL platform job runners from the ROS build farm . 
```

If you're using a target platform that has reached EOL, you should expect to stop receiving updated ROS packages.

> However, ROS Bosses may choose to perform one last update on a target platform after it reaches EOL.

Reviewer 3: the question that i have from user's perspective, what is the difference between distro E.O.L and platform E.O.L for us? this sounds like pretty much same for end users? if that is true, why dont we have just ROS 2 official E.O.L date (which includes platform E.O.L date) for users?

Author: > what is the difference between distro E.O.L and platform E.O.L for us? this sounds like pretty much same for end users? if that is true, why dont we have just ROS 2 official E.O.L date (which includes platform E.O.L date) for users?

The difference between the ROS distribution EOL date and the Platform EOL date is that the Platform EOL date is decided by the vendor of that platform (Debian, Canonical, Microsoft), and may be earlier than when the ROS distribution goes EOL. Historically the ROS distribution EOL date matches the Ubuntu platform EOL date.

Reviewer 3: if the platform EOL comes before ROS distro, that ROS distro on the platform also will be E.O.L, correct? (from this documentation) so for the end user's perspective, they only need to care about ROS E.O.L? i think this PR makes it clear that point.

Reviewer: I think the quick description here would be, "If a host OS goes EOL then that OS is no longer supported by the ROS Distro by default."

The real world example is Windows 10 going EOL for Humble.

Author: > if the platform EOL comes before ROS distro, that ROS distro on the platform also will be E.O.L, correct?

In my mind a ROS distro only has one EOL date. I wouldn't say that a ROS distro is EOL on a platform. I would only call a ROS distro EOL when we no longer support the ROS distro on any platforms.

What would you like this to say instead?

Reviewer 3: > a ROS distro only has one EOL date. I wouldn't say that a ROS distro is EOL on a platform.

i think this is okay, i was just checking if platform E.O.L prevails to ROS E.O.L

> However, ROS Bosses may choose to perform one last update on a target platform after it reaches EOL.

Reviewer 2: ```suggestion
However, ROS Bosses may choose to perform an update on a target platform after it reaches EOL in exceptional circumstances if the infrastructure is still available.
```

For ROS Bosses
--------------

Before a target platform reaches EOL:

* Make sure the ROS distribution documentation includes an EOL date for any target platform that reaches EOL before the ROS distribution.

> * When the EOL date for a target platform approaches, announce that ROS distribution will drop support for that target platform on Discourse.

Reviewer: We should give a recommendation here. Perhaps six months?

Author: That's a good idea. As a package maintainer, I think I'd find it helpful to have a heads up period measured in syncs, so I know when I need to make releases by in order to get changes or fixes onto a platform before it goes EOL.

Author: I put "at least 2 syncs" in 220325e5 . This would be a good timeframe to debate in the ROS meeting.

.. _PlatformEOLPolicy:

Platform EOL Policy
===================

.. contents:: Table of Contents
   :depth: 1
   :local:

:doc:`ROS distributions <Releases>` do not support end-of-life (EOL) platforms, even if the ROS distribution is still active.
This page explains:

* What users of EOL platforms should expect
* What ROS Bosses should do

Policy
------

Every ROS distribution supports certain **platforms**, such as Windows 11 or Ubuntu 24.04.

> **Vendors**, such as Microsoft or Canonical, decide how long they will support one of their platforms.

Reviewer: ```suggestion
**Vendors** of these platforms, such as Microsoft or Canonical, decide how long they will support one of their platforms.
```

Vendor is a bit jargony, this might help clarify.

When a vendor decides a platform has reached EOL, they usually stop publishing critical bug and security fixes.
To protect ourselves from potentially unpatched security vulnerabilities, we proactively remove all jobs on EOL platforms from the ROS build farm.

If you are using a platform that is no longer supported by its vendor, you should expect to stop receiving updated ROS packages.
Existing ROS packages will remain available and functional, but they will no longer be updated.
However, ROS Bosses may choose to update packages on an EOL platform in exceptional circumstances.

For ROS Bosses
--------------

Before a target platform reaches EOL:

* Make sure the ROS distribution documentation includes EOL dates for any platform that reaches EOL before the ROS distribution.

> * Post an announcement about the platform reaching EOL at least 2 syncs beforehand so that package maintainers have time to update their packages.

Reviewer: ```suggestion
* Post an announcement about the platform reaching EOL at least 2 syncs (roughly 60-90 days) beforehand so that package maintainers have time to update their packages.
```

A little more clarity. Sync frequency is a bit opaque to most users.
